src/protobuf.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _PROTOBUF_H
+#define _PROTOBUF_H
+
+#include "arch.h"
+#include <string.h>
+#include <type_traits>
+
+typedef const u32 protobuf_t;
+static protobuf_t VARINT = 0;
+static protobuf_t LEN = 2;
+
+typedef u32 protobuf_index_t;
+
+// We assume the length of a nested field can be represented with 3 varint
+// bytes.
```

Comment:
Can you please re-format the code according to the project's style? (indents, * pointers, line length, etc.)

Reply:
a11b31bcc0ae503aa3bbb688470bd863ffa84547

d7258c4ab618db1a731109c120bb3787d978ba92

Let me know if there's anything else I should change